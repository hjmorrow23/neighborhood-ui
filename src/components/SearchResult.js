import React from 'react';
import TopFind from './TopFind';
import $ from 'jquery';

const SearchResult = props => {
    function toggleTopFind(e) {
        e.preventDefault();
        const topFindContainer = document.getElementById("js-top-find");
        $(topFindContainer).toggleClass("result-top-finds-active");
    }
 
    return (
        <li className="result-item">
            <a href="" className="result-link">
                <span className="neighborhood-location">{props.neighborhood.name}, {props.neighborhood.city}, IL</span>
                <span className="result-type">Neighborhood</span>
                <span className="clear-right"></span>
            </a>
            <a href="" id="js-top-find-toggle" className="top-finds-link" onClick={toggleTopFind}>Top Finds</a>
            <div id="js-top-find" className="result-top-finds">
                {
                    props.neighborhood.topListings.map((listing, index) => 
                       <TopFind listing={listing} key={index}></TopFind> 
                    )
                }
            </div>
        </li>
    );
};



export default SearchResult;