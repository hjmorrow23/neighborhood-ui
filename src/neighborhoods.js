import img1 from "./assets/img/house1.jpg";
import img2 from "./assets/img/house2.jpg";
import img3 from "./assets/img/house3.jpg";
import img4 from "./assets/img/house4.jpg";
import img5 from "./assets/img/house5.jpg";
import img6 from "./assets/img/house6.jpg";

const neighborhoods = [
    {
        name: "Beverly",
        city: "Chicago",
        state: "IL",
        agent: "John Doe",
        overview: "Beverly is famous for its architecture, extensive green spaces, and historic charm. This South Side neighborhood is also known for its Irish-style pubs along Western Avenue, as well as its unique shopping district that stretches down 103rd Street. Beverly offers an eclectic environment like a big city, yet provides a distinct small-town vibe. The neighborhood has a variety of homes with plenty of backyard space for outdoor recreation.",
        rating: "5.0",
        starts: 5,
        topListings: [
            {
                address: "4 Privet Drive",
                price: "$375,000",
                image: img1
            },
            {
                address: "1060 West Addison Street",
                price: "$290,000",
                image: img2
            },
            {
                address: "1323 Mockingbird Lane",
                price: "310,000",
                image: img3
            },
        ]
    },
    {
        name: "Oakland",
        city: "Chicago",
        state: "IL",
        agent: "Jane Doe",
        overview: "Bordering the coast of Lake Michigan, Oakland is a quiet neighborhood on Chicago's south side. With scenic views of the glittering lake and Chicago's towering skyline, this promising neighborhood is sure to continue growing in the coming years.",
        rating: "5.0",
        starts: 5,
        topListings: [
            {
                address: "221B Baker Street",
                price: "$375,000",
                image: img4
            },
            {
                address: "425 Grove Street",
                price: "$290,000",
                image: img5
            },
            {
                address: "124 Conch Street",
                price: "310,000",
                image: img6
            },
        ]
    },
];

export default neighborhoods;