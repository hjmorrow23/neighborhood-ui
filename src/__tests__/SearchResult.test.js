import React, { Component } from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchResult from "../components/SearchResult";
import toJSON from 'enzyme-to-json';
import neighborhoods from '../neighborhoods';

configure({ adapter: new Adapter() });

const neighborhood = neighborhoods[0];

describe('<Dashboard />', () => {
    it('renders', () => {
        mount(
            <SearchResult neighborhood={neighborhood}/>
        );
    });

    it('matches the snapshot', () => {
        const wrapper = mount(
            <SearchResult neighborhood={neighborhood}/>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('updates via props', () => {
        const wrapper = mount(
            <SearchResult neighborhood={neighborhood} count={50} />
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.setProps({ count: 10 });
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
});
