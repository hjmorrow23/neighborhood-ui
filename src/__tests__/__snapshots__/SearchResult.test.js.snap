// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches the snapshot 1`] = `
<SearchResult
  neighborhood={
    Object {
      "agent": "John Doe",
      "city": "Chicago",
      "name": "Beverly",
      "overview": "Beverly is famous for its architecture, extensive green spaces, and historic charm. This South Side neighborhood is also known for its Irish-style pubs along Western Avenue, as well as its unique shopping district that stretches down 103rd Street. Beverly offers an eclectic environment like a big city, yet provides a distinct small-town vibe. The neighborhood has a variety of homes with plenty of backyard space for outdoor recreation.",
      "rating": "5.0",
      "starts": 5,
      "state": "IL",
      "topListings": Array [
        Object {
          "address": "4 Privet Drive",
          "image": "house1.jpg",
          "price": "$375,000",
        },
        Object {
          "address": "1060 West Addison Street",
          "image": "house2.jpg",
          "price": "$290,000",
        },
        Object {
          "address": "1323 Mockingbird Lane",
          "image": "house3.jpg",
          "price": "310,000",
        },
      ],
    }
  }
>
  <li
    className="result-item"
  >
    <a
      className="result-link"
      href=""
    >
      <span
        className="neighborhood-location"
      >
        Beverly
        , 
        Chicago
        , IL
      </span>
      <span
        className="result-type"
      >
        Neighborhood
      </span>
      <span
        className="clear-right"
      />
    </a>
    <a
      className="top-finds-link"
      href=""
      id="js-top-find-toggle"
      onClick={[Function]}
    >
      Top Finds
    </a>
    <div
      className="result-top-finds"
      id="js-top-find"
    >
      <TopFind
        key="0"
        listing={
          Object {
            "address": "4 Privet Drive",
            "image": "house1.jpg",
            "price": "$375,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            $375,000
          </p>
          <p
            className="top-find-address"
          >
            4 Privet Drive
          </p>
        </a>
      </TopFind>
      <TopFind
        key="1"
        listing={
          Object {
            "address": "1060 West Addison Street",
            "image": "house2.jpg",
            "price": "$290,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            $290,000
          </p>
          <p
            className="top-find-address"
          >
            1060 West Addison Street
          </p>
        </a>
      </TopFind>
      <TopFind
        key="2"
        listing={
          Object {
            "address": "1323 Mockingbird Lane",
            "image": "house3.jpg",
            "price": "310,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            310,000
          </p>
          <p
            className="top-find-address"
          >
            1323 Mockingbird Lane
          </p>
        </a>
      </TopFind>
    </div>
  </li>
</SearchResult>
`;

exports[`<Dashboard /> updates via props 1`] = `
<SearchResult
  count={50}
  neighborhood={
    Object {
      "agent": "John Doe",
      "city": "Chicago",
      "name": "Beverly",
      "overview": "Beverly is famous for its architecture, extensive green spaces, and historic charm. This South Side neighborhood is also known for its Irish-style pubs along Western Avenue, as well as its unique shopping district that stretches down 103rd Street. Beverly offers an eclectic environment like a big city, yet provides a distinct small-town vibe. The neighborhood has a variety of homes with plenty of backyard space for outdoor recreation.",
      "rating": "5.0",
      "starts": 5,
      "state": "IL",
      "topListings": Array [
        Object {
          "address": "4 Privet Drive",
          "image": "house1.jpg",
          "price": "$375,000",
        },
        Object {
          "address": "1060 West Addison Street",
          "image": "house2.jpg",
          "price": "$290,000",
        },
        Object {
          "address": "1323 Mockingbird Lane",
          "image": "house3.jpg",
          "price": "310,000",
        },
      ],
    }
  }
>
  <li
    className="result-item"
  >
    <a
      className="result-link"
      href=""
    >
      <span
        className="neighborhood-location"
      >
        Beverly
        , 
        Chicago
        , IL
      </span>
      <span
        className="result-type"
      >
        Neighborhood
      </span>
      <span
        className="clear-right"
      />
    </a>
    <a
      className="top-finds-link"
      href=""
      id="js-top-find-toggle"
      onClick={[Function]}
    >
      Top Finds
    </a>
    <div
      className="result-top-finds"
      id="js-top-find"
    >
      <TopFind
        key="0"
        listing={
          Object {
            "address": "4 Privet Drive",
            "image": "house1.jpg",
            "price": "$375,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            $375,000
          </p>
          <p
            className="top-find-address"
          >
            4 Privet Drive
          </p>
        </a>
      </TopFind>
      <TopFind
        key="1"
        listing={
          Object {
            "address": "1060 West Addison Street",
            "image": "house2.jpg",
            "price": "$290,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            $290,000
          </p>
          <p
            className="top-find-address"
          >
            1060 West Addison Street
          </p>
        </a>
      </TopFind>
      <TopFind
        key="2"
        listing={
          Object {
            "address": "1323 Mockingbird Lane",
            "image": "house3.jpg",
            "price": "310,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            310,000
          </p>
          <p
            className="top-find-address"
          >
            1323 Mockingbird Lane
          </p>
        </a>
      </TopFind>
    </div>
  </li>
</SearchResult>
`;

exports[`<Dashboard /> updates via props 2`] = `
<SearchResult
  count={10}
  neighborhood={
    Object {
      "agent": "John Doe",
      "city": "Chicago",
      "name": "Beverly",
      "overview": "Beverly is famous for its architecture, extensive green spaces, and historic charm. This South Side neighborhood is also known for its Irish-style pubs along Western Avenue, as well as its unique shopping district that stretches down 103rd Street. Beverly offers an eclectic environment like a big city, yet provides a distinct small-town vibe. The neighborhood has a variety of homes with plenty of backyard space for outdoor recreation.",
      "rating": "5.0",
      "starts": 5,
      "state": "IL",
      "topListings": Array [
        Object {
          "address": "4 Privet Drive",
          "image": "house1.jpg",
          "price": "$375,000",
        },
        Object {
          "address": "1060 West Addison Street",
          "image": "house2.jpg",
          "price": "$290,000",
        },
        Object {
          "address": "1323 Mockingbird Lane",
          "image": "house3.jpg",
          "price": "310,000",
        },
      ],
    }
  }
>
  <li
    className="result-item"
  >
    <a
      className="result-link"
      href=""
    >
      <span
        className="neighborhood-location"
      >
        Beverly
        , 
        Chicago
        , IL
      </span>
      <span
        className="result-type"
      >
        Neighborhood
      </span>
      <span
        className="clear-right"
      />
    </a>
    <a
      className="top-finds-link"
      href=""
      id="js-top-find-toggle"
      onClick={[Function]}
    >
      Top Finds
    </a>
    <div
      className="result-top-finds"
      id="js-top-find"
    >
      <TopFind
        key="0"
        listing={
          Object {
            "address": "4 Privet Drive",
            "image": "house1.jpg",
            "price": "$375,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            $375,000
          </p>
          <p
            className="top-find-address"
          >
            4 Privet Drive
          </p>
        </a>
      </TopFind>
      <TopFind
        key="1"
        listing={
          Object {
            "address": "1060 West Addison Street",
            "image": "house2.jpg",
            "price": "$290,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            $290,000
          </p>
          <p
            className="top-find-address"
          >
            1060 West Addison Street
          </p>
        </a>
      </TopFind>
      <TopFind
        key="2"
        listing={
          Object {
            "address": "1323 Mockingbird Lane",
            "image": "house3.jpg",
            "price": "310,000",
          }
        }
      >
        <a
          className="top-find"
          href=""
        >
          <img
            className="top-find-image"
            src="house1.jpg"
          />
          <p
            className="top-find-price"
          >
            310,000
          </p>
          <p
            className="top-find-address"
          >
            1323 Mockingbird Lane
          </p>
        </a>
      </TopFind>
    </div>
  </li>
</SearchResult>
`;
